todo-app/src/App.js

```diff
@@ -1,21 +1,56 @@
-import React, { Component } from 'react';
+import React from 'react';
 import logo from './logo.svg';
 import './App.css';
+import Todo from './Todo';
 
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
+  class App extends React.Component {
+    constructor (props){
+        super(props);
+        this.state = {
+          todos: [
+               {
+                description: "Todo List",
+                done: false
+               }
+            ]
+        };
+        
+    }
+    
+    handleChange = (e) => {
+    const description =  this.name.value;
+    const newToDo = {description};
+    const todosList = [...this.state.todos, newToDo];
+    this.setState({todos:todosList})
+  }
+    
+  
+   
+  onDelete= (index) => {
+    const todosList = [...this.state.todos];
+    todosList.splice(index, 1);
+    this.setState({todos:todosList});
+  }
+    render() {
+      let todoList=this.state.todos;
+      todoList= todoList.map(function(item,index){
+        return( <Todo  description={item.description} key={index} index={index} onDelete={this.onDelete}/>);
+      }.bind(this))
+      return (
+        <div className="App">
+          <div className="App-header">
+            <img src={logo} className="App-logo" alt="logo" />
+            <h2>To Do List</h2>
+          </div>
+          <div className= "App-Body">
+          {todoList}
+          </div>
+           <input ref={input => this.name = input} />
```

Comment:
Although there are special cases where `ref` can be used, in general we should [avoid using it](https://facebook.github.io/react/docs/refs-and-the-dom.html#dont-overuse-refs). To get and set the value of this input element, we should better use a [controlled component](https://facebook.github.io/react/docs/forms.html#controlled-components).